eqcorrscan/utils/pre_processing.py

```diff
@@ -999,18 +999,20 @@ def _prep_data_for_correlation(stream, templates, template_names=None,
                 template.traces[idx] for idx in stream_trace_id_dict[seed_id]])
             if len(template_channel) <= channel_index:
                 # out_template[channel_number].data = nan_channel  # quicker:
-                out_template[channel_number].__dict__['data'] = copy.deepcopy(
-                    nan_channel)
-                out_template[channel_number].stats.__dict__['npts'] = \
-                    template_length
-                out_template[channel_number].stats.__dict__['starttime'] = \
-                    template_starttime
-                out_template[channel_number].stats.__dict__['endtime'] = \
-                    UTCDateTime(ns=int(
+                out_template.traces[channel_number].__dict__[
+                    'data'] = copy.deepcopy(nan_channel)
```

Comment:
```suggestion
                    'data'] = np.copy(nan_channel)
```
I wondered if there was a faster numpy way to copy arrays, and there is (although not faster). On my machine:

```python
import numpy as np
import copy

a = np.random.randn(8640000)
%timeit copy.deepcopy(a)
# 11 ms ± 263 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
%timeit np.copy(a)
# 10 ms ± 166 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```
There is very little in it, but I can't see a good reason not to us `np.copy` (I just didn't know about it until now!).

Reply:
This is a very good point, should definitely contribute to the overall speedup. Thanks for finding this and merging!